src/solvers/smt2_incremental/convert_expr_to_smt.cpp

```diff
@@ -154,9 +154,22 @@ static smt_termt convert_expr_to_smt(const concatenation_exprt &concatenation)
 
 static smt_termt convert_expr_to_smt(const bitand_exprt &bitwise_and_expr)
 {
-  UNIMPLEMENTED_FEATURE(
-    "Generation of SMT formula for bitwise and expression: " +
-    bitwise_and_expr.pretty());
+  if(std::all_of(
+       bitwise_and_expr.operands().cbegin(),
+       bitwise_and_expr.operands().cend(),
+       [](exprt operand)
+       { return can_cast_type<integer_bitvector_typet>(operand.type()); }))
+  {
+    return convert_multiary_operator_to_terms(
+      bitwise_and_expr, smt_bit_vector_theoryt::make_and);
+  }
+  else
+  {
+    UNIMPLEMENTED_FEATURE(
+      "Bitwise \"AND\" conversion requires that all operands have the same "
+      "sort" +
```

Comment:
I can see your point about the wording, and of course I'm happy to change it to the previous wording inside the `UNIMPLEMENTED_FEATURE` macro, but I see no benefit to removing the else block.

To the contrary, I have a very keen preference for the `else` block to stay there, because it makes the structure of the code more obvious. 

Both with and without the `else` branch, the function handles two potential cases so far (one in which we can process the type of the input parameter, and one in which we don't). The presence of the `else` branch makes this fact explicit, revealing this hidden structure, compared to relying to the default behaviour of having the function fall through to an unimplemented feature macro to signify.

I'm open to changing this if there exist readability/technical issues that make the change you suggest superior to the existing structure.